coresimd/src/x86/i586/bmi.rs

```diff
@@ -163,8 +153,8 @@ pub unsafe fn _tzcnt_u64(x: u64) -> u64 {
 #[inline(always)]
 #[target_feature = "+bmi"]
 #[cfg_attr(test, assert_instr(tzcnt))]
-pub unsafe fn _mm_tzcnt_u32(x: u32) -> u32 {
-    x.trailing_zeros()
+pub unsafe fn _mm_tzcnt_32(x: u32) -> i32 {
```

Comment:
The renaming is correct, but these should return an unsigned integer (it returns a size). This integer is in range `[0,32)`, so per what we were currently doing somewhere else it should be an `u8`, but I have opened an issue to clarify that. 

Also, clang and GCC also define

```c++
 #define _tzcnt_u32(a)     (__tzcnt_u32((a)))
```

and in this module the non-`_mm` prefixed versions are missing (so the typo might have come from here).